js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * WeatherSystem manages what weather to spawn into a zone.
+ * 
+ * @author Caleb Krauter
+ */
+class WeatherSystem {
+
+    /**
+     * Pass in the type of weather, intensity and time of day.
+     * Direction of rain or snow is random.
+     * Rain or snow will always include clouds.
+     * Intensity is used for both rain/snow and clouds.
+     * 
+     * @param {String} type warm, hot, cloudy, rain, snow.
+     * @param {int} intensity a value between 0 and 5.
+     * @param {String} time day or night.
+     */
+    constructor(type, intensity, time) {
+        this.type = type;
+        this.intensity = intensity;
+        this.time = time;
+
+        // Rainy day or night
+        this.makeHeavens();
```

Comment:
While I like this method name... because it is great... It could be more clear. Generally, a well named method won't need a comment to describe it. 

Just food for thought.

Reply:
That is a good point. Would you say that the name is unclear, perhaps I'll just remove the comment. I suppose I could just name it addSunAndSky.

Reply:
Honestly, I don't think it needs to be renamed. I'm kind of on the fence about it. 

Probably in this case, since all the developers understand what it does, you could probably just omit the comment.